src/Honeycomb.Samplers/DeterministicSampler.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Security.Cryptography;
+using System.Text;
+using OpenTelemetry.Trace;
+
+namespace Honeycomb.Samplers
+{
+    public class DeterministicSampler : Sampler, IDisposable
+    {
+        private const string DescriptionFormat = "DeterministicSampler({0})";
+        private const string SampleRateAttributeName = "sampleRate";
+        private const string Hyphen = "-";
+        private const int Zero = 0;
+        private const int Four = 4;
+        private const int One = 1;
+        private const int Base16 = 16;
+        private readonly SHA1 sha1 = SHA1.Create();
+        private readonly int sampleRate;
+        private readonly long upperBound;

```

Comment:
I would probably make these readonly properties instead of fields, I think that's more in line with C# styles

Reply:
other than readonly generally using the underscore syntax, I prefer that syntax and it's more common in the code I've seen.

Reply:
Sure, I can convert to readonly properties.